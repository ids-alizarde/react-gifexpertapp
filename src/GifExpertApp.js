import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['Dragon Ball', 'Paw Patrol', 'Samurai X'];

    // const [categories, setCategories] = useState( ['Dragon Ball'] );
    const [categories, setCategories] = useState( defaultCategories );
    
    // const handleAdd = () => {

    //     // setCategories([ ...categories, 'Charly' ]);
    //     setCategories( cats => [ ...cats, 'Charly' ] );
    // };

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }></AddCategory>
            <hr/>

            <ol>
                {
                    // categories.map(category => {

                    //     return <li key = { category } >{ category }</li> 

                    // })

                    categories.map(category => (

                        <GifGrid 
                            key = { category }
                            category = { category }
                        />
                        
                    ))
                }
            </ol>
        </>
    )
}

