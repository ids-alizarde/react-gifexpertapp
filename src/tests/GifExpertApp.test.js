import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp'

describe('Pruebas en el componente <GifExpertApp />', () => {
    
    let wrapper = shallow( <GifExpertApp /> );

    test('Debe de mostrar correctamente el <GifExpertApp />', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });
    
    test('Debe de mostrar una lista de categorias', () => {

        const categories = [ 'Dragon Ball', 'Paw Patrol' ];
        wrapper = shallow( <GifExpertApp defaultCategories = { categories } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );
        
    })
    
});
