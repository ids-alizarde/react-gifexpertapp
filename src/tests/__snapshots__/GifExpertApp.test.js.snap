// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrar correctamente el <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Paw Patrol"
      key="Paw Patrol"
    />
  </ol>
</Fragment>
`;
